decidim-core/lib/tasks/upgrade/decidim_fix_nickname_uniqueness.rake

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+namespace :decidim do
+  namespace :upgrade do
+    desc "Modify nicknames with random numbers when exists similar ones case insensitively"
+    task fix_nickname_uniqueness: :environment do
+      logger = Logger.new($stdout)
+      logger.info("Updating users nickname ...")
+
+      # list of users already changed in the process
+      has_changed = []
+
+      Decidim::User.find_each do |user|
+        next if has_changed.include? user
+
+        Decidim::User.where("nickname ILIKE ?", user.nickname.downcase).order(:created_at).each do |similar_user|
+          next if has_changed.include? similar_user
+          next if user == similar_user
```

Comment:
Instead of checking for the same/changed users here, could you add this to the query above `.where.not(id: has_changed + [user.id])`?